automata/fa/nfa.py

```diff
@@ -631,3 +631,50 @@ def transition(states_pair, symbol):
                     pair_stack.append((r_1, r_2))
 
         return True
+
+    @classmethod
+    def levenshtein(cls, input_symbols, query_string, D):
+        """Code adapted from http://blog.notdot.net/2010/07/Damn-Cool-Algorithms-Levenshtein-Automata """
```

Comment:
Have the docstring also explain what is constructed and how the parameters affect the construction.
I think edit distance is the simplest term for it.